Rubberduck.Core/UI/UnitTesting/TestExplorerModel.cs

```diff
@@ -1,116 +1,97 @@
 using System;
-using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Linq;
 using System.Windows.Media;
 using System.Windows.Threading;
-using Rubberduck.Parsing.VBA;
+using Rubberduck.UI.UnitTesting.ViewModels;
 using Rubberduck.UnitTesting;
 using Rubberduck.VBEditor.SafeComWrappers.Abstract;
 
 namespace Rubberduck.UI.UnitTesting
 {
-    public class TestExplorerModel : ViewModelBase, IDisposable
+    internal class TestExplorerModel : ViewModelBase, IDisposable
     {
         private readonly IVBE _vbe;
-        private readonly RubberduckParserState _state;
         private readonly Dispatcher _dispatcher;
+        private readonly ITestEngine testEngine;
 
-        public TestExplorerModel(IVBE vbe, RubberduckParserState state)
+        public TestExplorerModel(IVBE vbe, ITestEngine testEngine)
         {
             _vbe = vbe;
-            _state = state;
-            _state.StateChanged += HandleStateChanged;
+            this.testEngine = testEngine;
 
+            testEngine.TestsRefreshed += RefreshTests;
+            testEngine.TestRunCompleted += HandleRunCompletion;
+            testEngine.TestCompleted += HandleTestCompletion;
```

Comment:
Why not `HandleTestsRefreshed`, `HandleTestRunCompleted`, and `HandleTestCompleted`? "test run" vs "test" could probably use some disambiguation, but I can't come up with better names ATM.